tools/gnur-runtime/src/main/java/org/renjin/gnur/api/Utils.java

```diff
@@ -76,13 +77,20 @@ public static void R_qsort_int_I(IntPtr iv, IntPtr II, int i, int j) {
     org.renjin.gnur.qsort.R_qsort_int_I(iv, II, i, j);
   }
 
-  @Deprecated
-  public static CharPtr R_ExpandFileName(CharPtr p0) {
-    throw new UnimplementedGnuApiMethod("R_ExpandFileName");
-  }
-
   public static BytePtr R_ExpandFileName(BytePtr p0) {
-    throw new UnimplementedGnuApiMethod("R_ExpandFileName");
+    if (p0.nullTerminatedStringLength() < 2 || p0.getChar(0) != '~' || Character.isAlphabetic(p0.getChar(1))) {
```

Comment:
This should call out to `Files.pathExpand` so that the logic is defined in exactly one place. For example:

```.java
  return BytePtr.nullTerminatedString(Files.pathExpand(Stdlib.nullTerminatedString(ptr)));
```